apps/microsoft-teams/frontend/src/components/config/NotificationsSection/NotificationsSection.tsx

```diff
@@ -53,13 +55,65 @@ const NotificationsSection = (props: Props) => {
     });
   };
 
-  const updateNotification = (index: number, editedNotification: Partial<Notification>) => {
+  const updateNotification = (
+    index: number,
+    editedNotification: Partial<Notification>,
+    isNew?: boolean
+  ) => {
     const notificationsPayload = [...notifications];
-    notificationsPayload[index] = { ...notificationsPayload[index], ...editedNotification };
-    dispatch({
-      type: actions.UPDATE_NOTIFICATIONS,
-      payload: notificationsPayload,
-    });
+    const existingNotification = notificationsPayload[index];
+
+    // Update the notification at the specified index
+    const updatedNotification = { ...existingNotification, ...editedNotification };
+    notificationsPayload[index] = updatedNotification;
+
+    // Check if the updated notification is unique
+    const uniqueNotifications = getUniqueNotifications(notificationsPayload);
+    const isUnique = uniqueNotifications.length === notificationsPayload.length;
+
+    // If not unique, open the modal and set the index for further handling
+    if (!isUnique) {
+      ModalLauncher.open(({ isShown, onClose }) => {
+        return (
+          <DuplicateModal
+            isShown={isShown}
+            handleCancel={() => {
+              onClose(true);
+            }}
+            handleConfirm={() => {
+              if (isNew) {
```

Comment:
If the duplicate notification is new and the user chooses to go to the existing duplicate notification, then this new one will go away. If it's an existing notification that is editing to be a duplicate of another, then that notification will reset back to its initial state when you navigate to the other notification.